backend/baseclass.pm

```diff
@@ -586,7 +586,13 @@ sub reset_consoles {
     # we iterate through all consoles
     for my $console (keys %{$testapi::distri->{consoles}}) {
         next if $self->console($console)->{args}->{persistent};
-        $self->reset_console({testapi_console => $console});
+        try {
+            local $SIG{__DIE__} = 'DEFAULT';
+            $self->reset_console({testapi_console => $console});
+        }
+        catch {
+            return {error => $_};
```

Comment:
The `return` will return from the catch function. I think you want to return from the `sub reset_consoles` here? You could use `eval` to avoid this problem.

Reply:
I would be open for pointers how to actually test this. Probably mock a dying reset console call?

Reply:
I guess I would go for the usual mocking and it it to `23-baseclass.t`.